overlord/servicestate/quota_handlers.go

```diff
@@ -220,17 +220,8 @@ func ensureSnapServicesForGroup(st *state.State, grp *quota.Group, opts *ensureS
 			return err
 		}
 
-		// drop disabled services from the startup ordering
-		startupOrderedMinusDisabled := make([]*snap.AppInfo, 0, len(startupOrdered)-len(disabledSvcs))
-
-		for _, svc := range startupOrdered {
-			if !isDisabledSvc[svc.ServiceName()] {
-				startupOrderedMinusDisabled = append(startupOrderedMinusDisabled, svc)
-			}
-		}
-
 		st.Unlock()
-		err = wrappers.RestartServices(startupOrderedMinusDisabled, nil, meter, perfTimings)
+		err = wrappers.RestartServices(startupOrdered, nil, nil, meter, perfTimings)
```

Comment:
I'm a bit surprised you didn't need to change any unit tests here, but I can look at expanding the quota tests in a followup after this lands, this code as-is looks correct, though you could also now get rid of all the code above Querying the disabled services since we don't need that anymore